src/main/kotlin/kotlinx/kover/KoverPlugin.kt

```diff
@@ -43,27 +43,22 @@ class KoverPlugin : Plugin<Project> {
     }
 
     private fun Project.createCollectingTask() {
-        tasks.create(COLLECT_TASK_NAME, Copy::class.java) {
+        tasks.create(COLLECT_TASK_NAME, KoverCollectingTask::class.java) {
             it.group = VERIFICATION_GROUP
             it.description = "Collects reports from all submodules in one directory."
-            allprojects { sub ->
-                val xmlReportTask = sub.tasks.withType(KoverXmlReportTask::class.java).getByName(XML_REPORT_TASK_NAME)
+            it.outputDir.set(project.layout.buildDirectory.dir("reports/kover/all"))
+
+            allprojects { p ->
```

Comment:
`project` instead of `p` looks like a more meaningful name here

Reply:
Then the parameter clashes the `Project.project` property - it's confusing.

Reply:
`subproject` or at least `proj` then